libres/lib/analysis/enkf_linalg.cpp

```diff
@@ -29,8 +28,8 @@ void enkf_linalg_genX3(matrix_type *X3, const matrix_type *W,
         for (j = 0; j < nrobs; j++)
             matrix_iset(X1, i, j, eig[i] * matrix_iget(W, j, i));
 
-    matrix_matmul(X2, X1, D); /*  X2 = X1 * D */
-    matrix_matmul(X3, W, X2); /*  X3 = W * X2 = X1 * X2 */
+    *X2 = *X1 * *D;
```

Comment:
I would guess this could be something like:
```C++
EigenMatrix X1(nrmin, nrobs);
// init X1

auto X2 = X1 * *D;
```
to reduce the dereferencing and manual memory management a bit. With a bit more refactoring the `X3` matrix could also be returned by value instead of using an output argument.

Reply:
Yes. But we have separate tickets for refactoring these functions so I figured that I'd avoid doing too much now.

Reply:
This refactor is currently underway in: #3018